.github/workflows/reusable-build-push.yml

```
@@ -3,71 +3,89 @@ name: Reusable workflow to build and push multi-arch images
+          else
+            PREV_VERSION=0.0.0
+          fi
+          NEW_VERSION=$(git show HEAD:$ROS2_WS_PATH/VERSION.${{ inputs.ros_distro }})
+
+          echo "prev_version=${PREV_VERSION}" >> $GITHUB_OUTPUT
+          echo "new_version=${NEW_VERSION}" >> $GITHUB_OUTPUT
+        shell: bash
+      - uses: aica-technology/.github/.github/actions/compare-versions@v0.11.0
+        id: check
+        with:
+          previous_version: ${{ steps.versions.outputs.prev_version }}
+          new_version: ${{ steps.versions.outputs.new_version }}
+
   metadata:
+    needs: check-version
+    if: ${{ needs.check-version.outputs.has_changed == 'true' || inputs.force }}
     runs-on: ubuntu-latest
-    name: Parse inputs
+    name: Metadata
     outputs:
-      output_tag: ${{ steps.parse.outputs.output_tag }}
-      image_name: ${{ steps.parse.outputs.image_name }}
-      path: ${{ steps.parse.outputs.path }}
-      file: ${{ steps.parse.outputs.file }}
-      build_flags: ${{ steps.parse.outputs.build_flags }}
+      image_name: ${{ steps.ensure-image.outputs.image_name }}
+      image_tags: ${{ steps.tags.outputs.image_tags }}
+      build_flags: ${{ steps.tags.outputs.build_flags }}
+      git_tag: ${{ steps.tags.outputs.git_tag }}
     steps:
-      - name: Parse inputs
-        id: parse
+      - uses: aica-technology/.github/.github/actions/ghcr-ensure-prefix@v0.6.0
+        id: ensure-image
+        with:
+          image_name: aica-technology/ros2-ws
+
+      - id: tags
         run: |
-          BUILD_FLAGS=()
-          # parse base tag     
-          BASE_TAG=${{ inputs.base_tag }}
+          BUILD_FLAGS=()    
+          BASE_TAG=${{ inputs.ros_distro }}
           BUILD_FLAGS+=(--build-arg BASE_TAG=${BASE_TAG})
           echo "::debug::Using base image tag ${BASE_TAG}"
-          # parse output tag
-          OUTPUT_TAG=${{ inputs.output_tag }}
-          if [ -z ${OUTPUT_TAG} ]; then
-            OUTPUT_TAG=${BASE_TAG}
-          fi
-          echo "output_tag=${OUTPUT_TAG}" >> $GITHUB_OUTPUT
-          echo "::debug::Generated image tag will be ${OUTPUT_TAG}"
-          
-          # parse image name
-          IMAGE_NAME=${{ inputs.workspace }}
-          IMAGE_NAME=${{ github.repository_owner }}/${IMAGE_NAME//[_]/-}
-          echo "::debug::Generated image name will be ${IMAGE_NAME}"
-          echo "image_name=${IMAGE_NAME}" >> $GITHUB_OUTPUT
-          
-          # parse docker path and file
-          WS_PATH=${{ inputs.workspace }}
-          WS_PATH=${WS_PATH//[-]/_}
-          echo "path=${WS_PATH}" >> $GITHUB_OUTPUT
-          FILE=${WS_PATH}/Dockerfile${{ inputs.dockerfile_extension }}
-          echo "file=${FILE}" >> $GITHUB_OUTPUT
-
           echo "build_flags=${BUILD_FLAGS[@]}" >> $GITHUB_OUTPUT
 
+          VERSION_TAG="v${{ needs.check-version.outputs.version }}"
+          IMAGE_TAGS="${{ inputs.ros_distro}}-${VERSION_TAG}","${{ inputs.ros_distro}}"
```

Shouldn't this be:
```suggestion
          IMAGE_TAGS="${VERSION_TAG},${VERSION_TAG}-${{ inputs.ros_distro}}","${{ inputs.ros_distro}}"
```
So we get `1.0.0`, `1.0.0-iron` and `iron`?

---

I'm a bit afraid to have tags without the ROS distro in it. What if we have other distros at some point?

---

We should never reuse the tags even between distros anyway. What's worrying you?

---

Well if we always include the distro with the version, then we could potentially have a `1.0.0-jazzy` no? Given your statement it seems not though, so we can commit your suggestion if you prefer.

---

Jazzy would be a breaking change so we would go to `2.0.0-jazzy` right? For me the distro name is just a convenience for clarity (thus why we should have been using `+iron` but it's not possible so that's the next best thing).